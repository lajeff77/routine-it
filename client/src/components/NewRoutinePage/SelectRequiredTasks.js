import React, {useEffect, useState} from 'react';
import {Box, Card, Typography, Checkbox, FormGroup, FormControlLabel} from '@mui/material';
import {Formik, Form} from 'formik';

const SelectRequiredTasks = ({props}) => {
    const [checkedItems, setCheckedItems] = useState([]);

    let formik = props.formik;

    useEffect(() => {
        formik.validateForm();
    },[])

    useEffect(() => {
        switch(props.id){
            case 0:
                formik.setFieldValue("lowEffortRoutine", checkedItems);
                break;
            case 1:
                formik.setFieldValue("medEffortRoutine", checkedItems);                
                break;
            case 2:
                formik.setFieldValue("highEffortRoutine", checkedItems);
                break;
            default:
                //do nothing
                break;
        }
    }, [checkedItems]);

    const handleChange = (event) => {
        const { name, checked } = event.target;
        if (checked) {
            setCheckedItems(prevItems => [...prevItems, name]);
        } else {
            setCheckedItems(prevItems => prevItems.filter(item => item !== name));
        }
    };

    return(
        <Box sx={{margin:"20px", display:"flex", flexDirection:"column", alignItems:"center"}}>
            <Typography variant="h1" sx={{textAlign: 'center', margin: '30px'}}>{props.text.title}</Typography>
            <Card variant="outlined" sx={{align:"center", width:"70%", display:"flex", flexDirection:"column", outline:"2px"}}>
                <Typography variant="body1"  sx={{margin: '20px'}}>{props.text.prompt}</Typography>
                <Box>
                    <Formik initialValues={{list: checkedItems}} validateOnChange={true}>
                        <Form name="list">
                            {formik.values.tasks.map((item)=> {return(
                                <Box>
                                    <FormGroup sx={{margin:"20px"}}>
                                        <FormControlLabel control={<Checkbox name={item} checked={checkedItems.includes(item)} onChange={handleChange}/>} label={item} />
                                    </FormGroup>
                                </Box>)
                            })}
                        </Form>
                    </Formik> 
                 </Box>   
            </Card>
        </Box>
    )
}

export default SelectRequiredTasks;

