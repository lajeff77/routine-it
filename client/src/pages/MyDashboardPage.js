import React from "react";
import Dashboard from "../components/Dashboard";
import PermanentSideDrawer from "../components/PermanentSideDrawer";
import {Grid, Item} from '@mui/material';

const MyDashboardPage = () => {

    return (
        <div>
            <Grid container rowSpacing={2} >
                <Grid item xs={2}><PermanentSideDrawer /></Grid>
                <Grid item xs={10}><Dashboard /></Grid>
            </Grid>
        </div>
    )
}

export default MyDashboardPage;