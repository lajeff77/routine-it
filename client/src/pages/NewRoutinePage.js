import React, {useState} from 'react';
import {Box} from '@mui/material';
import NameRoutine from '../components/NameRoutine';
import SurveyNav from '../components/SurveyNav';

const NewRoutinePage = () => {
    const [index,setIndex] = useState(0);
    const survey =[<NameRoutine/>];

    return(
        <Box>
            {survey[index]}
            <SurveyNav props={{setIndex:setIndex, index:index, maxIndex:survey.length-1}}/>
        </Box>
    );
}

export default NewRoutinePage;