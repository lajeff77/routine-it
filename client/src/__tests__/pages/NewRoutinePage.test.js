import { render} from "@testing-library/react";
import React from "react";
import NewRoutinePage from "../../pages/NewRoutinePage";
import MyDashboardPage from "../../pages/MyDashboardPage";
import { MemoryRouter, Route, Routes } from "react-router-dom";



describe(NewRoutinePage.name, () => { 
    test('NewRoutinePage matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
            <Routes>
                <Route element={<MyDashboardPage />} path="/mydashboard" />
                <Route element={<NewRoutinePage />} path="/newroutine" />
            </Routes>
            </MemoryRouter>
        );
    
        expect(asFragment()).toMatchSnapshot();
    });
});