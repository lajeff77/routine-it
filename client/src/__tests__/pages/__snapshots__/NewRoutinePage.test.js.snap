// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRoutinePage NewRoutinePage matches snapshot 1`] = `<DocumentFragment />`;
